cookbook/rag_examples/rag_example.py

```diff
@@ -0,0 +1,84 @@
+"""A script using RAG to summarize relevant articles from a large database in 2 ways.
+
+News articles from 2004 are embedded so they may be queried (in two ways) according to a
+topic of choice. In the first way we manually calculate similarity between query and
+embeddings. In the second, we set up Pinecone and efficiently use its search to find the
+most relevant articles. In both cases, we use the Mirascope library to handle the logic
+so as to streamline prompt creation.
+"""
+import os
+
+import pandas as pd
+from config import FILENAME, MAX_TOKENS, URL
+from rag_prompts.news_rag_prompt import LocalNewsRagPrompt, PineconeNewsRagPrompt
+from setup_pinecone import setup_pinecone
+from utils import embed_df_with_openai, load_data
+
+from mirascope import OpenAIChat
+
+chat = OpenAIChat(api_key=os.getenv("OPENAI_API_KEY"))
+
+
+# Make reruns efficient by preprocessing data before saving to csv
+if not os.path.exists(FILENAME):
```

Comment:
I would refactor this into a separate setup script:

```sh
$ python setup_local.py
$ python setup_pinecone.py
```

Then in this file I would defer checking this stuff until you check the flags (see below comment):

```python
def main():
    ...
    if flags.use_pinecone:
        if not pinecone_is_setup:
            raise ValueError("`use_pinecone` specified but pinecone is not setup.")
        ...
    else:
        if not os.path.exists(FILENAME):
            raise ValueError("You must first run `setup_local.py`.")
        df = pd.read_pickle(FILENAME)
        ...
```

Reply:
Also see below comment. Is there some way to figure out `pinecone_is_setup`? If not easily through pinecone, could we do something like save a dummy file `pinecone_is_setup.txt` and then check `if not os.path.exists("pinecone_is_setup.txt")